File: Top 100 Codes/07-Codes related to Matrix/04-Find row with maximum no. of 1’s/Findrowwithmaximumnoof1s.cpp
#include <iostream>
#include <vector>
using namespace std;

// Function to find the row with the maximum number of 1s
int rowWithMax1s(vector<vector<int>>& mat) {
    int n = mat.size();
    int m = mat[0].size();

    int maxRowIndex = -1;
    int j = m - 1; // Start from top-right

    for (int i = 0; i < n; i++) {
        while (j >= 0 && mat[i][j] == 1) {
            j--;
            maxRowIndex = i;
        }
    }

    return maxRowIndex;
}

int main() {
    vector<vector<int>> mat = {
        {0, 0, 0, 1},
        {0, 1, 1, 1},
        {0, 0, 1, 1},
        {0, 0, 0, 0}
    };

    int index = rowWithMax1s(mat);
    if (index != -1)
        cout << "Row with maximum 1s is: " << index << endl;
    else
        cout << "No 1s in the matrix" << endl;

    return 0;
}

File: Top 100 Codes/07-Codes related to Matrix/04-Find row with maximum no. of 1’s/Findrowwithmaximumnoof1s.py
def row_with_max_1s(matrix):
    n = len(matrix)
    m = len(matrix[0])

    max_row_index = -1
    j = m - 1  # Start from top-right corner

    for i in range(n):
        while j >= 0 and matrix[i][j] == 1:
            j -= 1
            max_row_index = i

    return max_row_index

# Test input
mat = [
    [0, 0, 0, 1],
    [0, 1, 1, 1],
    [0, 0, 1, 1],
    [0, 0, 0, 0]
]

index = row_with_max_1s(mat)
if index != -1:
    print(f"Row with maximum 1s is: {index}")
else:
    print("No 1s in the matrix")
